addons/mail/controllers/main.py

```diff
@@ -200,6 +210,14 @@ def mail_action_view(self, model=None, res_id=None, access_token=None, **kwargs)
 
             models that have an access_token may apply variations on this.
         """
+        if model and res_id:
+            record_mapping = self.upgrade_record_mapping(model, res_id)
```

Comment:
Shouldn't we instead have localized fixes in account to handle real models we want to support instead of adding queries for all models / all links ?

Reply:
It will be a source of bugs.
Maybe some cache can be added, but I don't think it's worth it.

Reply:
But you add queries for links that anyway have a short life. Feels like a corner case is impacting all valid use cases.